docs/guide/docker.md

```diff
@@ -0,0 +1,114 @@
+---
+id: docker
+title: 🚢 Build docker image
+description: Build simple docker image that you can use on the production
+sidebar_position: 8
+---
+
+After we have completed the development of our application, the moment comes when we want to upload it to the real world, most often we use Docker for this, wrapping our application in a container.
+
+To do this we need to follow a few simple steps:
+
+## Step 1 - Setup project
+
+First create folder for our project run the following command in the console:
+
+```bash title="~/"
+mkdir example && cd example
+```
+
+Then we have to init go project:
+```bash title="~/example"
+go mod init example.org/demo
+```
+
+Add **Fiber** to our project:
+```bash title="~/example"
+go get github.com/gofiber/fiber/v2
+```
+
+## Step 2 - Hello world
+
+To start we shold create `main.go` file:
+```bash title="~/example"
+touch main.go
+```
+
+Next we should add following content to our file:
+
+```go title="~/app/main.go"
+package main
+
+import "github.com/gofiber/fiber/v2"
+
+func main() {
+	app := fiber.New()
+
+	app.Get("/", func(c *fiber.Ctx) error {
+		return c.SendString("Hello, World!")
+	})
+
+	app.Listen(":4000")
+}
+```
+
+## Step 3 - create Dockerfile
+
+Run the following command in the console:
+
+```bash
+touch Dockerfile
+```
+
+For the image we will use [Distroless](https://github.com/GoogleContainerTools/distroless) docker image, in the file we should add following content:
+
+```bash title="~/example/Dockerfile"
+FROM golang:1.21-alpine as base
+
+WORKDIR /app
+COPY . .
+
+RUN go mod download && go mod verify
+RUN GOOS=linux GOARCH=amd64 go build -o /server .
+
+FROM gcr.io/distroless/static-debian12
```

Comment:
You are using Alpine for the based which uses `musl` and debian for the final stage which uses `glibc`. This can causr major issues depending on your application. The way to do this is:

```Dockerfile
FROM golang:1.21 as build

WORKDIR /go/src/app
COPY . .

RUN go mod download
RUN go vet -v
RUN go test -v

RUN CGO_ENABLED=0 go build -o /go/bin/app

FROM gcr.io/distroless/static-debian11

COPY --from=build /go/bin/app /
CMD ["/app"]
```

Reply:
fixed